flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -485,6 +494,23 @@ public void requestPartitions() throws IOException, InterruptedException {
 		}
 	}
 
+	@Override
+	public void blockInputChannel(int channelIndex) {
+		InputChannel inputChannel = indexToInputChannelMap.get(channelIndex);
+		if (inputChannel == null) {
+			throw new IllegalStateException("Could not find input channel from the channel index " + channelIndex);
```

Comment:
`IllegalArgumentException`?

Reply:
I referred to the other similar usage in `IllegalStateException`. If existing this condition that `channelIndex` is correct but the mapping construction is wrong.

I think `IllegalArgumentException` also makes sense.

Reply:
both argumentations sound right:
how about `checkArgument(0 <= channelIndex && channelIndex < numberOfInputChannels)` and `checkState(inputChannel != null)`